src/main/webapp/dev/js/common/questionRubric.es6

```diff
@@ -10,6 +10,110 @@ import {
+        || (!isSwapLeft && $swapCell.next(rubricCellSelector).length === 0)) {
+        // trying to swap left most or right most column
+        return;
+    }
+
+    const swapValues = ($a, $b) => {
+        const temp = $a.val();
+        $a.val($b.val());
+        $b.val(temp);
+    };
+
+    const numberOfRows = parseInt($(`#rubricNumRows-${questionNum}`).val(), 10);
+    let swapColIndex;
+
+    if (isSwapLeft) {
+        swapColIndex = $swapCell.prev(rubricCellSelector).attr('data-col');
+    } else {
+        swapColIndex = $swapCell.next(rubricCellSelector).attr('data-col');
+    }
+
+    // swap rubric choices
+    let $currentValue = getRubricChoiceElem(questionNum, colIndex);
+    let $swapValue = getRubricChoiceElem(questionNum, swapColIndex);
+    swapValues($currentValue, $swapValue);
+
+    // swap rubric weights
+    $currentValue = getRubricWeightElem(questionNum, colIndex);
+    $swapValue = getRubricWeightElem(questionNum, swapColIndex);
+    swapValues($currentValue, $swapValue);
+
+    // swap options filled
+    for (let row = 0; row < numberOfRows; row += 1) {
+        $currentValue = getRubricDescElem(questionNum, row, colIndex);
+        $swapValue = getRubricDescElem(questionNum, row, swapColIndex);
+        swapValues($currentValue, $swapValue);
+    }
+}
+
+function moveRubricCol(questionNum, colIndex, isMoveLeft) {
+    if ($(`#rubricEditTable-${questionNum}`).length === 0
+            || $(`.rubricCol-${questionNum}-${colIndex}`).length === 0
+            || typeof isMoveLeft !== 'boolean') {
+        // question and column should exist, isSwapLeft must be boolean
+        return;
+    }
+
+    swapRubricCol(questionNum, colIndex, isMoveLeft);
+}
+
+function disableCornerMoveRubricColumnButtons(questionNum) {
+    const $optionColumns = $(`#rubric-options-row-${questionNum} td[class*='rubricCol-']`);
+
+    const checkMoveLeftOfFirstCol = () => {
+        const $leftmostCol = $optionColumns.first();
+        const leftmostColIndex = $leftmostCol.attr('data-col');
+        const $leftmostColLeftBtn = $leftmostCol.find(`#rubric-move-col-left-${questionNum}-${leftmostColIndex}`);
+
+        $leftmostColLeftBtn.prop('disabled', true);
+    };
+
+    const checkMoveRightOfLastCol = () => {
+        const $rightmostCol = $optionColumns.last();
+        const rightmostColIndex = $rightmostCol.attr('data-col');
+        const $rightmostColRightBtn = $rightmostCol.find(`#rubric-move-col-right-${questionNum}-${rightmostColIndex}`);
+
+        $rightmostColRightBtn.prop('disabled', true);
+    };
+
+    const checkMoveRightOfSecondLastCol = () => {
+        const $secondlastCol = $optionColumns.last().prev();
+        const secondlastColIndex = $secondlastCol.attr('data-col');
+        const $secondlastColRightBtn = $secondlastCol.find(`#rubric-move-col-right-${questionNum}-${secondlastColIndex}`);
+
+        $secondlastColRightBtn.prop('disabled', false);
+    };
```

Comment:
> why is the second last column's right button treated specially?

For a special case -
  - If a new column is added, then 2nd last column's right button would be disabled, it must be enabled.

> There's some repetition here, which can be refactored out.

Where is the repetition?

>  Perhaps calling it setCornerButtonState would be better.

Renaming them to `disableMoveLeftOfFirstCol`, `disableMoveRightOfLastCol`, `enableMoveRightOfSecondLastCol`